controllers/apiv3/api_base_controller.py

```diff
@@ -91,7 +91,7 @@ def post(self, *args, **kw):
         self._validate_tba_auth_key()
         self._errors = ValidationHelper.validate_request(self)
         if self._errors:
-            self.abort(400)
+            self.abort(404)
```

Comment:
@fangeugene What's your reasoning? It doesn't make sense to me because a `400` means "bad request". For example, 400 is returned [here](https://github.com/SUPERCILEX/the-blue-alliance/blob/fae50437048384a6a12b8929c2f4d95e41d491e4/controllers/apiv3/api_base_controller.py#L132) where it makes sense because the auth header is missing, hence a "bad request".

However, if a team, event, etc. doesn't exist, it's not a bad request, but instead a `404` which means "Not Found Error". That makes way more sense because it's what's actually happening: the team, event, etc. wasn't found or doesn't exist.